go/cmd/dolt/commands/merge.go

```diff
@@ -136,30 +138,65 @@ func (cmd MergeCmd) Exec(ctx context.Context, commandStr string, args []string,
 	// check for fast-forward merge
 	rows, err := sql.RowIterToRows(sqlCtx, schema, rowIter)
 	if err != nil {
+		cli.Println("merge successful, but failed to check for fast-forward")
 		cli.Println(err.Error())
-		return 1
+		return 0
 	}
 	if rows != nil && rows[0][1].(int64) == 1 {
 		cli.Println("Fast-forward")
 	}
 
 	// calculate merge stats
 	if !apr.Contains(cli.AbortParam) {
-		if apr.Contains(cli.NoCommitFlag) && apr.Contains(cli.NoFFParam) {
+		mergeHash, err := getHashOf(queryist, sqlCtx, apr.Arg(0))
+		if err != nil {
+			cli.Println("merge successful, but failed to get hash of merge ref")
+			cli.Println(err.Error())
+			return 0
+		}
+		headHash, err := getHashOf(queryist, sqlCtx, "HEAD")
+		if err != nil {
+			cli.Println("merge successful, but failed to get hash of HEAD")
+			cli.Println(err.Error())
+			return 0
+		}
+		cli.Println("Updating", headHash+".."+mergeHash)
+
+		if apr.Contains(cli.NoCommitFlag) {
 			cli.Println("Automatic merge went well; stopped before committing as requested")
 		}
 
 		mergeStats := make(map[string]*merge.MergeStats)
+		mergeStats, noConflicts, ok := calculateMergeConflicts(queryist, sqlCtx, mergeStats)
+		if !ok {
+			return 0
+		}
 
-		mergeStats, noConflicts, err := calculateMergeConflicts(queryist, sqlCtx, mergeStats)
-		if err != 0 {
-			return 1
+		if noConflicts {
+			if apr.Contains(cli.NoCommitFlag) {
+				mergeStats, ok = calculateMergeStats(queryist, sqlCtx, mergeStats, "STAGED")
```

Comment:
This is switched. As discussed, want HEAD -> STAGED

Add test to ensure we get the directionality right here.